SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorPackageAssembly.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKEditorPackageAssembly.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+#include "Misc/App.h"
+#include "Misc/FileHelper.h"
+#include "Misc/MonitoredProcess.h"
+#include "UnrealEdMisc.h"
+
+#include "SpatialGDKEditorModule.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKServicesConstants.h"
+#include "SpatialGDKServicesModule.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKEditorPackageAssembly);
+
+#define LOCTEXT_NAMESPACE "SpatialGDKEditorPackageAssembly"
+
+namespace
+{
+	const FString SpatialBuildExe = FSpatialGDKServicesModule::GetSpatialGDKPluginDirectory(TEXT("SpatialGDK/Binaries/ThirdParty/Improbable/Programs/Build.exe"));
+	const FString Linux = TEXT("Linux");
+	const FString Win64 = TEXT("Win64");
+} // anonymous namespace
+
+FSpatialGDKPackageAssembly::FSpatialGDKPackageAssembly()
+{
+}
+
+void FSpatialGDKPackageAssembly::LaunchTask(const FString& Exe, const FString& Args, const FString& WorkingDir)
+{
+	PackageAssemblyTask = MakeShareable(new FMonitoredProcess(Exe, Args, WorkingDir, /* Hidden */ true));
+	PackageAssemblyTask->OnCompleted().BindSP(this, &FSpatialGDKPackageAssembly::OnTaskCompleted);
+	PackageAssemblyTask->OnOutput().BindSP(this, &FSpatialGDKPackageAssembly::OnTaskOutput);
+	PackageAssemblyTask->OnCanceled().BindSP(this, &FSpatialGDKPackageAssembly::OnTaskCanceled);
+	PackageAssemblyTask->Launch();
+}
+
+void FSpatialGDKPackageAssembly::BuildAssembly(const FString& ProjectName, const FString& Platform, const FString& Configuration, const FString& AdditionalArgs)
+{
+	FString WorkingDir = FPaths::ConvertRelativePathToFull(FPaths::ProjectDir());
+	FString Project = FPaths::ConvertRelativePathToFull(FPaths::GetProjectFilePath());
+	FString Args = FString::Printf(TEXT("%s %s %s %s %s"), *ProjectName, *Platform, *Configuration, *Project, *AdditionalArgs);
+	LaunchTask(SpatialBuildExe, Args, WorkingDir);
```

Comment:
Does this call result in the same set of operation that are taking place when packaging the project from the editor ? Because if it does, we may want to redirect it to the packaging commands, which may have better/more familiar UI.

Reply:
There are additional steps we do in the Build script before calling the UAT.